### PR TITLE
Add middleware for redirecting URLs in the site

### PR DESCRIPTION
**Pull request by emersonlaurentino**
Redirect any docs or component to developer portal keeping the path

### CODE REVIEW

File: apps/site/middleware.ts [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+        '/docs',
+        ''
+      )}`
+    )
+  }
+  if (pathname.startsWith('/components')) {
+    return NextResponse.redirect(
+      `https://developers.vtex.com/docs/guides/faststore${pathname
+        .replace('/components', '')
+        .replace(/\/([^\/]*)\//, '/$1-')}`
```
Comment (eduardoformiga) [inline apps/site/middleware.ts:17]: what does this regex do?

**Review by eduardoformiga — COMMENTED** [review]
Have we already migrated all the docs?

File: apps/site/middleware.ts (hunk 1 above)
Comment (emersonlaurentino) [reply apps/site/middleware.ts:17]: faststore.dev uses `/components/atoms/checkbox` but the developer portal uses `/atoms-checkbox`. So, this regex is responsible for update only the second `/`

Comment (emersonlaurentino) [reply apps/site/middleware.ts:17]: you can test here, using the preview https://faststore-site-git-redirect-site-faststore.vercel.app/components/atoms/checkbox

**Pull request merged**